compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -141,12 +133,11 @@ impl<'a> Parser<'a> {
                     }
                 }
             }
-            NonterminalKind::Pat => {
-                let (mut pat, tokens) = self.collect_tokens(|this| match or_pat_mode {
-                    OrPatNonterminalMode::TopPat => {
-                        this.parse_top_pat(GateOr::Yes, RecoverComma::No)
-                    }
-                    OrPatNonterminalMode::NoTopAlt => this.parse_pat(None),
+            pat_kind @ (NonterminalKind::Pat2018 | NonterminalKind::Pat2021) => {
```

Comment:
```suggestion
            NonterminalKind::Pat2018 | NonterminalKind::Pat2021 => {
```
`kind` is already available.